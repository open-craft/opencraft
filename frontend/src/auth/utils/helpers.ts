import { V2Api } from 'global/api';

export const checkAuthAndRefreshToken = async () => {
  const refreshToken = localStorage.getItem('token_refresh');
  let response = null;

  if (refreshToken) {
    try {
      response = await V2Api.authRefreshCreate({
        data: { refresh: refreshToken }
      });
    } catch (e) {
      return false;
    }
  }

  if (response) {
    localStorage.setItem('token_access', response.access);
    return true;
  }

  return false;
};
