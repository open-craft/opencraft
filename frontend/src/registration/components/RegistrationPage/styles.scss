@import "~styles/theme";

.registration-page {
  display: flex;
  flex-direction: column;
  align-items: stretch;

  .step-bar-container {
    background-color: $primary-1;
    margin: 0px -15px;
  }

  h1, h2 {
    text-align: center;
    color: #fff;
  }

  h1 {
    font-family: $playfair-font;
    font-weight: bold;
    font-size: 40px;

    &.big {
      font-size: 50px;
    }
  }

  h2 {
    font-family: $chivo-font;
    font-size: 20px;
    margin-top: 20px;
    margin-bottom: 4rem;
  }

  .registration-page-container {
    background-color: #fff;
    padding: 4rem 1rem;
    margin: 0 -15px;
    flex: 1;

    .registration-page-content {
      margin: 0 auto;
      max-width: 700px;
      justify-content: center;
    }
  }
}
