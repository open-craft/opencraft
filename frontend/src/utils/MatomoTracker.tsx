import React, { ReactNode } from 'react';
import {
  createInstance,
  MatomoProvider,
  useMatomo
} from '@datapunt/matomo-tracker-react';
import {
  MATOMO_ALIAS_DOMAIN,
  MATOMO_BASE_URL,
  MATOMO_MY_DOMAIN,
  MATOMO_SITE_ID
} from '../global/constants';

interface MatomoTrackerProps {
  children: ReactNode;
}

interface MatomoAdditionalConfiguration {
  [propName: string]: any;
}

interface MatomoConfiguration {
  urlBase: string;
  siteId: number;
  configurations?: MatomoAdditionalConfiguration;
}

interface MatomoUserIdTrackerProps {
  userId: string;
}

/*
 * This wrapper component initializes the Matomo tracking by wrapping the children
 * elements, if the tracking is enabled by providing a valid value for the 'MATOMO_BASE_URL'
 * constant. It allows tracking page views, events and the user id, when available.
 *
 * The 'MATOMO_ALIAS_DOMAIN' constant allows the cross-domain linking of another site with
 * this site so that they are treated as the same site in Matomo.
 */
export const MatomoTracker: React.FC<MatomoTrackerProps> = (
  props: MatomoTrackerProps
) => {
  let element = props.children;
  if (MATOMO_BASE_URL) {
    const matomoConfiguration: MatomoConfiguration = {
      urlBase: MATOMO_BASE_URL,
      siteId: parseInt(MATOMO_SITE_ID, 10)
    };
    if (MATOMO_ALIAS_DOMAIN) {
      const myDomain = `*.${MATOMO_MY_DOMAIN}`;
      const aliasDomain = `*.${MATOMO_ALIAS_DOMAIN}`;
      matomoConfiguration.configurations = {
        enableCrossDomainLinking: true,
        setDomains: [myDomain, aliasDomain]
      };
    }
    element = (
      <MatomoProvider value={createInstance(matomoConfiguration)}>
        {props.children}
      </MatomoProvider>
    );
  }
  return <>{element}</>;
};

/**
 * This is a component passes the provided user id to Matomo if Matomo tracking is enabled.
 * The 'pushInstruction' function exposed by the 'useMatomo()' hook is used to do this.
 *
 * This component is needed as all the components in the registration process are classes and hence
 * the 'yseMatomo' hook cannot be directly used in those components.
 */
export const MatomoUserIdTracker: React.FC<MatomoUserIdTrackerProps> = (
  props: MatomoUserIdTrackerProps
) => {
  const { pushInstruction } = useMatomo();
  if (MATOMO_BASE_URL && props.userId) {
    pushInstruction('setUserId', props.userId);
  }
  return null;
};
