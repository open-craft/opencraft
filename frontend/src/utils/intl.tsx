import React from "react";
import { FormattedMessage, MessageDescriptor } from "react-intl";

type MessageMap = { [k: string]: Omit<MessageDescriptor, 'id'> };

interface Props<T, K extends keyof T> {
    id: K;
    messages: T;
    values?: any;
}


export const WrappedMessage = (props: Props<MessageMap, string>) => {
    const {messages, id, ...values} = props;
    return (
        <FormattedMessage  {...messages[id]} id={id} {...values}/>
    );
};
