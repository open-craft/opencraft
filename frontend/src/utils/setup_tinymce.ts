import 'tinymce/tinymce';
import 'tinymce/themes/silver';
import 'tinymce/skins/ui/oxide/skin.min.css';
import 'tinymce/skins/ui/oxide/content.min.css';
import 'tinymce/plugins/advlist';
import 'tinymce/plugins/autolink';
import 'tinymce/plugins/lists';
import 'tinymce/plugins/link';
import 'tinymce/plugins/image';
import 'tinymce/plugins/charmap';
import 'tinymce/plugins/print';
import 'tinymce/plugins/preview';
import 'tinymce/plugins/anchor';
import 'tinymce/plugins/searchreplace';
import 'tinymce/plugins/visualblocks';
import 'tinymce/plugins/code';
import 'tinymce/plugins/fullscreen';
import 'tinymce/plugins/insertdatetime';
import 'tinymce/plugins/media';
import 'tinymce/plugins/table';
import 'tinymce/plugins/paste';
import 'tinymce/plugins/help';
import 'tinymce/plugins/wordcount';
