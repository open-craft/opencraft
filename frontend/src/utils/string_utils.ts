export const toCamelCase = (s: string) => {
  return s.replace(/([-_][a-z])/gi, ($1) => {
    return $1.toUpperCase().replace('-', '').replace('_', '');
  });
};

export const sanitizeErrorFeedback = (input: {
  [key: string]: Array<string>;
}) => {
  // Loop at each error message and join them.
  // Also convert keys from snake_case to camelCase
  const newObject: any = {};

  Object.keys(input).forEach((key) => {
    newObject[toCamelCase(key)] = input[key].join();
  });

  return newObject;
};

export const capitalizeFirstLetter = (str: string) => {
  return str[0].toUpperCase() + str.slice(1);
};
