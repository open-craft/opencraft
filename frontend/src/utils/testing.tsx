import { createMemoryHistory } from 'history';
import * as React from 'react';
import { IntlProvider } from 'react-intl';
import { Provider } from 'react-redux';
import { MemoryRouter } from 'react-router';
import * as renderer from 'react-test-renderer';
import { applyMiddleware, createStore } from 'redux';
import thunk from 'redux-thunk';
import { createRootReducer } from '../global/reducers';

export const setupComponentForTesting = (
  reactContent: JSX.Element,
  storeContents = {}
) => {
  const middleware = applyMiddleware(thunk);

  const store = createStore(
    createRootReducer(createMemoryHistory()),
    storeContents,
    middleware
  );

  return renderer.create(
    <IntlProvider textComponent={React.Fragment} locale="en">
      <Provider store={store}>
        <MemoryRouter>{reactContent}</MemoryRouter>
      </Provider>
    </IntlProvider>
  );
};
