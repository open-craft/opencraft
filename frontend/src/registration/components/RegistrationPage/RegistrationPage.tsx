import * as React from 'react';
import { Row } from 'react-bootstrap';
import './styles.scss';
import { StepBar } from 'ui/components/StepBar';

interface Props {
  title: string;
  subtitleBig?: string;
  subtitle?: string;
  children: React.ReactNode;
  currentStep: number;
}

export const RegistrationPage: React.FC<Props> = (props: Props) => {
  return (
    <div className="registration-page">
      <h1>{props.title}</h1>
      {props.subtitleBig && <h1 className="big">{props.subtitleBig}</h1>}
      {props.subtitle && <h2>{props.subtitle}</h2>}

      <div className="step-bar-container">
        <div className="d-flex justify-content-center">
          <StepBar count={4} currentStep={props.currentStep} />
        </div>
      </div>

      <div className="registration-page-container">
        <Row className="registration-page-content">{props.children}</Row>
      </div>
    </div>
  );
};
