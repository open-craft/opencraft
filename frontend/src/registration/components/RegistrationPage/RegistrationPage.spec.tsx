import React from 'react';
import { setupComponentForTesting } from 'utils/testing';
import { RegistrationPage } from './RegistrationPage';

it('renders without crashing', () => {
  const tree = setupComponentForTesting(<RegistrationPage />).toJSON();
  expect(tree).toMatchSnapshot();
});
