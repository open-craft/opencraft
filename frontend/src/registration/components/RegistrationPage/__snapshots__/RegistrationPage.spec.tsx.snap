// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders without crashing 1`] = `
<div
  className="registration-page"
>
  <h1 />
  <div
    className="step-bar-container"
  >
    <div
      className="d-flex justify-content-center"
    >
      <div
        className="step-bar"
      >
        <svg
          className="step"
          viewBox="0 0 10 10"
          xmlns="http://www.w3.org/2000/svg"
        >
          <circle
            cx={5}
            cy={5}
            fill="#6d9cae"
            r={5}
          />
        </svg>
        <span
          className="line"
        />
        <svg
          className="step"
          viewBox="0 0 10 10"
          xmlns="http://www.w3.org/2000/svg"
        >
          <circle
            cx={5}
            cy={5}
            fill="#6d9cae"
            r={5}
          />
        </svg>
        <span
          className="line"
        />
        <svg
          className="step"
          viewBox="0 0 10 10"
          xmlns="http://www.w3.org/2000/svg"
        >
          <circle
            cx={5}
            cy={5}
            fill="#6d9cae"
            r={5}
          />
        </svg>
        <span
          className="line"
        />
        <svg
          className="step"
          viewBox="0 0 10 10"
          xmlns="http://www.w3.org/2000/svg"
        >
          <circle
            cx={5}
            cy={5}
            fill="#6d9cae"
            r={5}
          />
        </svg>
        <span
          className="line"
        />
      </div>
    </div>
  </div>
  <div
    className="registration-page-container"
  >
    <div
      className="registration-page-content row"
    />
  </div>
</div>
`;
